lib/datadog/core/telemetry/metric.rb

```diff
@@ -0,0 +1,126 @@
+# frozen_string_literal: true
+
+module Datadog
+  module Core
+    module Telemetry
+      module Metric
+        # Base class for Metric
+        class Base
+          class << self
+            def request_type
+              raise NotImplementedError
+            end
+          end
+
+          attr_reader :values
+
+          def initialize(name, tags)
+            @name = name
+            @tags = tags
+            @values = nil
+          end
+
+          def update_value(value)
+            raise NotImplementedError
+          end
+
+          def metric_type
+            raise NotImplementedError
+          end
+
+          def timestamp
+            Time.now.to_i
+          end
+
+          def to_h
+            {
+              tags: @tags.map { |k, v| "#{k}:#{v}" },
+              values: @values,
+              type: metric_type,
+              common: true,
+            }
+          end
+        end
+
+        # GenerateMetricType
+        class GenerateMetricType < Base
+          class << self
+            def request_type
+              'generate-metrics'
+            end
+          end
+        end
+
+        # DistributionsMetricType
```

Comment:
Minor: It seems weird to add these comments just to "shut up" rubocop -- we're actually making our code worse to make rubocop "happy".